learn/how-tos/integrating-server-side-pagination.md

```diff
@@ -0,0 +1,187 @@
+```
+
+```
+Sample Response:
+{
+	data: [...{}],
+_meta: {
+	$limit: 5,
+	$offset: 5,
+	itemCount: 30,
+hasMoreItems: true 
+}
+}
+```
+
+## Page
+
+Page divides set into pages. The API accepts a page parameter of an integer type, indicating specific data the page needs to return. In this pagination, you can jump to any page, retrieve data on that particular page, and make parallel requests.
+
+### Configuring Page
+
+The following the code snippet describes how to configure the Page pagination.
+
+- Create a [Service Variable](/learn/app-development/variables/web-service#how-to-create-a-service-variable).
+- Drag-n-drop a data widget; for example, Data table.
+- Bind the Data Table to Service Variable.
+- Go to Script tab, configure Page pagination as described below. 
+
+```js
+{
+   type: ‘page’,
+   input: {
+       page: 'page',
+       size: ‘size’
+   },
+   output: {
+       totalElements: '$body.totalPassengers'           
+  }
+}
+```
+
+### Example
+
+API: https://sampleapi.com?size=10&page=1
+
+```js
+Sample Response:
+{
+	data: [...{}],
+totalPassengers: 73760,
+totalPages: 7376
+}
+```
+
+## Cursor Pagination
+
+Cursor works by returning a pointer to a specific item in the set of data. On subsequent requests, the database returns results after the given pointer.
+
+### Configuring Cursor
+
+The following the code snippet describes how to configure the Cursor pagination.
+
+- Create a [Service Variable](/learn/app-development/variables/web-service#how-to-create-a-service-variable).
+- Drag-n-drop a data widget; for example, Data table.
+- Bind the Data Table to Service Variable.
+- Go to Script tab, configure Cursor pagination as described below. 
+
+```js
+{
+   type: ‘cursor’,
+   input: {
+       page: 'page'
+   },
+   output: {
+       next: '$header.X-WM-X-Links-Next',
+       prev: '$header.X-WM-X-Links-Previous'
+   }
+}
+```
+
+### Example
+
+API: https://sampleapi.com/v2/users?page=1
+
+```js
+Sample Response: [.....{}]
+Sample Response Headers:
+
+x-links-current: https://gorest.co.in/public/v2/users?page=1
+x-links-next: https://gorest.co.in/public/v2/users?page=2
+x-links-previous: 
+x-pagination-limit: 20
+x-pagination-page: 1
+x-pagination-pages: 138
+x-pagination-total: 2751
+```
+
+## JMES Library
+
+While providing values for keys to the input and output parameters, you must follow JMES guidelines.
+
+JMESPath  allows you to declaratively specify how to extract elements from a JSON document. It is a query language for JSON. Define paths using JMES as below. Using “jmespath” library to evaluate JMES expressions. For more information on JMES visit the following link:  https://jmespath.org/
+
+### Example
+
+**Response headers**:
+
+```
+{	   "status":{"message":[{"message_TYPE":"SU","messageDesc":"SUCCESS","messageCode":"0000"}]},
+"X-WM-pagination":{"totalRecordCount":"75","nextStartIndex":"11","hasMoreRecords":"Y","numRecReturned":"2"}}
+```
+
+**hasMoreItems**: '$header."X-WM-pagination".hasMoreRecords == Y'
```

Comment:
This seem to have abruptly ended. May be prefix with.

Pagination controls can be sent both as body of a HTTP Request or a HTTP Header. Here we show how to send these in a HTTP header.